Source/Utils/WarpXMovingWindow.cpp

```diff
@@ -246,20 +253,103 @@ WarpX::shiftMF (MultiFab& mf, const Geometry& geom, int num_shift, int dir,
     shiftiv[dir] = num_shift;
     Dim3 shift = shiftiv.dim3();
 
+    const RealBox& real_box = geom.ProbDomain();
+    const auto dx = geom.CellSizeArray();
+    std::unique_ptr<ParserWrapper> field_parsewrap;
+    bool useparser = false;
+
+    if (ext_init == true && B_flag == true) {
+        if (B_ext_grid_s == "constant" || B_ext_grid_s == "default")
+            external_field = B_external_grid[dir];
+        if (B_ext_grid_s == "parse_b_ext_grid_function") {
+           useparser = true;
+           if (dir==0)
+              field_parsewrap.reset(new ParserWrapper(
+                              makeParser(str_Bx_ext_grid_function)));
```

Comment:
When the moving window is ON, are you constructing a new `ParserWrapper` each time the window moves (e.g., roughly at each iteration)? For performance reasons, I think it would be best to avoid that, and store the parser in `WarpX` class.

Reply:
Thank you so much for pointing this out. I have now fixed this